flink-libraries/flink-streaming-python/src/main/java/org/apache/flink/streaming/python/api/environment/PythonStreamExecutionEnvironment.java

```diff
@@ -0,0 +1,442 @@
+		this.env = StreamExecutionEnvironment.createRemoteEnvironment(host, port, jar_files);
+		this.registerJythonSerializers();
+	}
+
+	private PythonStreamExecutionEnvironment(String host, int port, Configuration config, String... jar_files) {
+		this.env = StreamExecutionEnvironment.createRemoteEnvironment(host, port, config, jar_files);
+		this.registerJythonSerializers();
+	}
+
+	private PythonStreamExecutionEnvironment(String host, int port, int parallelism, String... jar_files) {
+		this.env = StreamExecutionEnvironment.createRemoteEnvironment(host, port, parallelism, jar_files);
+		this.registerJythonSerializers();
+	}
+
+	private void registerJythonSerializers() {
+		this.env.registerTypeWithKryoSerializer(PyString.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyInteger.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyLong.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyUnicode.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyTuple.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyObjectDerived.class, PyObjectSerializer.class);
+		this.env.registerTypeWithKryoSerializer(PyInstance.class, PyObjectSerializer.class);
+	}
+
+	public PythonDataStream create_python_source(SourceFunction<Object> src) throws Exception {
+		return new PythonDataStream<>(env.addSource(new PythonGeneratorFunction(src)).map(new UtilityFunctions.SerializerMap<>()));
+	}
+
+	/**
+	 * Add a java source to the streaming topology. The source expected to be an java based
+	 * implementation (.e.g. Kafka connector).
+	 *
+	 * @param src  A native java source (e.g. PythonFlinkKafkaConsumer09)
+	 * @return Python data stream
+	 */
+	public PythonDataStream add_java_source(SourceFunction<Object> src) {
+		return new PythonDataStream<>(env.addSource(src).map(new UtilityFunctions.SerializerMap<>()));
+	}
+
+	/**
+	 * A thin wrapper layer over {@link StreamExecutionEnvironment#fromElements(java.lang.Object[])}
+	 *
+	 * @param elements
+	 * 		The array of PyObject elements to create the data stream from.
+	 * @return The data stream representing the given array of elements
+	 */
+	public PythonDataStream from_elements(PyObject... elements) {
+		return new PythonDataStream<>(env.fromElements(elements));
+	}
+
+	/**
+	 * A thin wrapper layer over {@link StreamExecutionEnvironment#fromCollection(java.util.Collection)}
+	 *
+	 * <p>The input {@code Collection} is of type {@code Object}, because it is a collection
+	 * of Python elements. * There type is determined in runtime, by the Jython framework.</p>
+	 *
+	 * @param collection
+	 * 		The collection of python elements to create the data stream from.
+	 * @return
+	 *     The data stream representing the given collection
+	 */
+	public PythonDataStream from_collection(Collection<Object> collection) {
```

Comment:
It concerns the Jython framework. Apparently, when using argument expansion (as in the case of `from_elements`), the Jython framework instantiates PyObject for each element in the the elements. On the other hand, when using a Collection, the Jython framework, instantiates a `PyList` with Java native classes (.e.g Integer, String, etc).